src/model_repository_manager.cc

```diff
@@ -476,20 +498,20 @@ ModelRepositoryManager::PollAndUpdateInternal(bool* all_models_polled)
 
   infos_.swap(new_infos);
 
-  UpdateDependencyGraph(added, deleted, modified);
+  UpdateDependencyGraph(&dependency_graph_, infos_, added, deleted, modified);
 
   for (const auto& name : deleted) {
     model_life_cycle_->AsyncUnload(name);
   }
 
   // model loading / unloading error will be printed but ignored
-  LoadModelByDependency();
+  LoadModelByDependency(false /* release_mu */);
```

Comment:
In case the poll frequency is too fast (poll every 1 second and it takes 2 seconds to poll), there will be parallel poll if the mutex is released. This is to prevent this scenario.

Reply:
`LoadModelByDependency()` no longer require the mutex to be held, after some logic refactoring.
Update commit: [Refactor parallel load logic](https://github.com/triton-inference-server/core/pull/165/commits/0fa4e296b613806d0bd7de77775f09e51d866d21)